### PR TITLE
Attached clusters data sources

### PR DESCRIPTION
**Pull request by hankfreund**
* Added new datasource for retrieving valid attached cluster platform versions `google_container_attached_versions`
* Added new datasource for retrieving a manifest to apply to the user cluster for boot-strapping `google_container_attached_install_manifest`

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-datasource
google_container_attached_versions
```

```release-note:new-datasource
google_container_attached_install_manifest
```

### CODE REVIEW

**Comment by ScottSuarez** [comment]
Hey ! wanted to check in. Is there a corresponding issues for these datasources being added? Looking for more information on the need and reason for why. internal or external

**Comment by hankfreund** [comment]
Hi! Sorry for the confusion. These datasources round out the required implementation for the new attached clusters API. You can see my previous PR for adding the resource here: https://github.com/GoogleCloudPlatform/magic-modules/pull/6858
There is not a corresponding external issue for this. We can chat offline if there's any other info you need.

File: mmv1/third_party/terraform/data_sources/data_source_google_container_attached_install_manifest.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"manifest": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
```
Comment (ScottSuarez) [inline mmv1/third_party/terraform/data_sources/data_source_google_container_attached_install_manifest.go:36]: Switching from "parent" to `project`, `region` might be a better UX. If region is identical to location this `region` can be omitted.

File: mmv1/templates/terraform/examples/container_attached_cluster_basic.tf.erb [hunk 2]
```diff
@@ -1,6 +1,11 @@
 data "google_project" "project" {
```
Comment (ScottSuarez) [inline mmv1/templates/terraform/examples/container_attached_cluster_basic.tf.erb:1]: Both data sources could use a datasource markdown file.

Checkout https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/website/docs/d

File: mmv1/templates/terraform/examples/container_attached_cluster_basic.tf.erb [hunk 3]
```diff
@@ -1,6 +1,11 @@
...
 }
```
Comment (ScottSuarez) [inline mmv1/templates/terraform/examples/container_attached_cluster_basic.tf.erb:2]: I would prefer splitting this down to two PRs.. one for each data source. To keep the context more focused. It doesn't seem like the data sources are dependent on eachother.

**Review by ScottSuarez — CHANGES_REQUESTED** [review]
Seems fairly straightforward.. Here are some initial thoughts.

**Comment by hankfreund** [comment]
I've split the PR into two. This one is now only for `google_container_attached_versions` and test modifications to use it.
Addressed the review comments.

**Pull request merged**